enumerables.rb

```diff
@@ -0,0 +1,243 @@
+      yield(h_array[v])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    h_array = to_a
+    0.upto(length - 1) do |v|
+      yield(h_array[v], v)
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    h_array = to_a
+    new_arr = []
+    var = 0
+    h_array.my_each do |v|
+      if yield v
+        new_arr[var] = v
+        var += 1
+      end
+    end
+    new_arr
+  end
+
+  def my_all?(num = nil)
+    h_array = to_a
+
+    h_array.my_each do |v|
+      break unless num.nil? == false
+      return false unless (v == num) == true
+    end
+    return true unless block_given?
+
+    unless block_given?
+      h_array.my_each do |v|
+        return false unless v == true
+      end
+    end
+    return true unless block_given?
+
+    h_array.my_each do |v|
+      return false unless yield(v)
+    end
+    true
+  end
+
+  def my_any?
+    h_array = to_a
+    h_array.my_each do
+      return true unless yield(v) == false
+    end
+    false
+  end
+
+  def my_none?(num = nil)
+    h_array = to_a
+
+    h_array.my_each do |v|
+      break unless num.nil? == false
+      return false unless (v == num) == false
+    end
+    return true unless block_given?
+
+    h_array.my_each do |v|
+      return false unless yield(v) == false
+    end
+    true
+  end
+
+  def my_count(num = '')
+    h_array = to_a
+    count = 0
+    h_array.my_each do
+      count += 1
+    end
+    return count unless (num == '') == false || block_given?
+
+    num.to_i
+    count = 0
+    h_array.my_each do |v|
+      count += 1 unless (v == num) == false
+    end
+    return count unless block_given?
+
+    count = 0
+    h_array.my_each do |v|
+      count += 1 unless yield(v) == false
+    end
+    count
+  end
+
+  def my_map
+    return to_enum(:double) unless block_given?
+
+    h_array = to_a
+    map_arr = []
+
+    h_array.my_each do |v|
+      map_arr.push(yield(v))
+    end
+    map_arr
+  end
+
+  def my_inject(num = 0, &block)
+    return 'no block given (LocalJumpError)' unless block_given? || num != 0
+
+    sum = 0
+    h_range = to_a
+    h_range.my_each do |v|
+      sum += v
+    end
+    return sum unless (num == :+) == false
+
+    sum = 1
+    h_range.my_each do |v|
+      sum *= v
+    end
+    return sum unless (num == :*) == false
+
+    sum = 0
+    h_range.my_each do |v|
+      sum += v
+    end
+    return sum unless !num.zero? || block_given?
+
+    sum = num
+    h_range.my_each do |v|
+      sum += v
+    end
+    return sum unless block_given?
+
+    sum = num
+    sum = h_range[0] unless num.positive?
+    index = 0
+    h_range.my_each do
+      sum = block.yield(sum, h_range[index]) unless num.positive?
+      sum = block.yield(sum, h_range[index - 1]) unless num.positive? == false
+      index += 1
+      break unless (index == h_range.length) == false
+    end
+    return sum unless num.positive?
+
+    sum
+  end
+end
```

Comment:
- **_#my_inject raises a "LocalJumpError" when no block or argument is given_**
- **_This example mentioned in the last review still failing:  `#my_inject` searches for the longest word in an array of strings
`['to','four','tri'].my_inject(&proc{|a,b| [a,b].max_by{|m| m.length}}) == ['to','four','tri'].inject(&proc{|a,b| [a,b].max_by{|m| m.length}})`_**
- **_`#my_inject`, when a block is given without an initial value, combines all elements of enum by applying a binary operation, specified by a block_**
- **_`#my_inject`, when a symbol is specified with an initial value, combines each element of the collection by applying the symbol as a named method_**
- **_`#my_inject`, when a symbol is specified with an initial value, combines each element of the collection by applying the symbol as a named method::range_**